lib/msf/core/post/file.rb

```diff
@@ -501,19 +501,21 @@ def rm_rf(*remote_dirs)
   alias :dir_rm :rm_rf
 
   #
-  # Rename a remote file.
+  # Renames a remote file and returns true on success and false
+  # on failure
   #
   # @param old_file [String] Remote file name to move
   # @param new_file [String] The new name for the remote file
   def rename_file(old_file, new_file)
+    verification_token = Rex::Text.rand_text_alphanumeric(8)
```

Comment:
Consider exploring using the error code from each cmd_exec for comparison instead of a token.

https://stackoverflow.com/questions/334879/how-do-i-get-the-application-exit-code-from-a-windows-command-line
https://stackoverflow.com/questions/11004666/returning-an-error-code-from-cmd-to-powershell

Reply:
Thanks for the links will go through and try to use the error code 🙂

Reply:
removed `&&` and now using `$?` to verify the success as `&&` might not be present for powershell version <7.